### PR TITLE
bump to chakra 3

### PR DESCRIPTION
**Pull request by dtemkin1**
Bumps version of Chakra UI, mainly followed [migration guide](https://www.chakra-ui.com/docs/get-started/migration) (which is the reason for the components/ui folder) but also made some of the following changes:

- Manually set the variant of buttons to "subtle" to match how Hydrant already looks like, same with setting every link component to blue
- Using component select component instead of native select. Whenever the theme was dark the options were all unreadable before :(
- For some reason when we clicking an event on the calendar the click event isn't firing? Seems to be getting constantly re-rendered so refactored the Calendar.tsx file a bit to try things out but couldn't get it working consistently. If someone better at React knows whats going on pls lmk :/
- Prettier touched a lot of files, sorry for the massive diff

If I missed something, let me know! :D

### CODE REVIEW

File: src/components/ui/avatar.tsx
Comment (cjquines) [inline src/components/ui/avatar.tsx:1]: **suggestion:** let's remove the files here that are unused?

File: src/lib/utils.tsx [hunk 1]
```diff
@@ -85,15 +88,18 @@ export function sum(arr: Array<number>): number {
...
 
 export function urldecode(obj: string): any {
   return Msgpack.decode(
     new Uint8Array(
       atob(obj)
         .split("")
-        .map((c) => c.charCodeAt(0))
-    )
+        .map((c) => c.charCodeAt(0)),
+    ),
```
Comment (cjquines) [inline src/lib/utils.tsx:103]: **thought:** we should actually put prettier in the project, #33

File: src/components/ClassTable.scss
Comment (cjquines) [inline src/components/ClassTable.scss:1]: **question:** what are these changes?

Comment (dtemkin1) [reply src/components/ClassTable.scss:1]: these were mainly me trying to get it to look similar to https://www.chakra-ui.com/docs/components/table

File: src/components/ui/avatar.tsx
Comment (dtemkin1) [reply src/components/ui/avatar.tsx:1]: should be resolved with commit commit adc6805a60c68ca2edbe36bd68f2ff6aa233b4dc

File: src/lib/utils.tsx (hunk 1 above)
Comment (psvenk) [reply src/lib/utils.tsx:103]: I've now opened #85; we could merge that first and then rebase to make the diff here more manageable.

Comment (dtemkin1) [reply src/lib/utils.tsx]: rebased with main @psvenk

**Review by psvenk — COMMENTED** [review]
This mostly looks good to me, code-wise and functionality-wise. I have a few comments, though:

- As you noted, the click event doesn't consistently fire when clicking on a class/activity on the calendar; strangely, this issue also exists for the buttons in the description of a custom activity (for changing its properties).
- In general, I have mixed feelings about how this affects the general design language used in Hydrant. I personally prefer the more defined features of the old design; in particular, what are your thoughts on bringing back the smaller spaces between selected classes/activities, larger font size for selected class/activity names, and even thicker font weights and rounder corners?

For reference:

Before:
hydrant mit edu_

After:
localhost_3000_

**Comment by dtemkin1** [comment]
I made a few style changes to make it look more like it did before, let me know if that helps. The two things that I'm struggling with (and maybe someone with better React knowledge can help with this?) and should be fixed before merging are:

- [ ] The text being shown by a tooltip component is appearing below other things (eg. buttons, the table header, etc.). I'm not sure if we have to do things with refs or something, but if someone knows how to fix this (and has a bit more time than me to work on this), please let me know!
- [ ] The calendar component seems to be rerendering events constantly, which is making it so that clicking events doesn't work (and also just making the website slower I think). I poked around and it seems that React thinks that the props given to the Calendar component (specifically, selectedActivities) is constantly changing. I tried to revert most things that touch the Calendar component back to how they were originally, but I'm not really sure why this is happening.

**Review by psvenk — APPROVED** [review]

**Pull request merged**